vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/Response.java

```diff
@@ -1,209 +1,58 @@
 /*
- * Copyright (C) 2017 Julien Viet
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
  *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
  */
 
 package io.vertx.pgclient.impl.codec;
 
+import io.vertx.codegen.annotations.VertxGen;
+
 /**
  *
  * <p>
  * A common response message for PostgreSQL
  * <a href="https://www.postgresql.org/docs/9.5/static/protocol-error-fields.html">Error and Notice Message Fields</a>
- *
- * @author <a href="mailto:emad.albloushi@gmail.com">Emad Alblueshi</a>
  */
+@VertxGen
+public interface Response {
```

Comment:
I think we don't actually need this API, we can put the fields in the `PgException`. In addition I'm not sure if all the fields are necessary to be exposed directly in the API.

Reply:
agreeing with you, let's have fields on exception